src/pseudopeople/configuration/validator.py

```diff
@@ -171,3 +188,21 @@ def _validate_probability(
             base_error_message + f"'{parameter}'s must be between 0 and 1 (inclusive). "
             f"Provided {noise_type_config}."
         )
+
+
+def _validate_nickname_probability(
+    noise_type_config: Union[int, float], parameter: str, base_error_message: str
+) -> None:
+    _validate_probability(noise_type_config, parameter, base_error_message)
+    if noise_type_config > metadata.NICKNAMES_PROPORTION:
+        logger.warning(
+            base_error_message
+            + f"Maximum configuration value is {1/metadata.NICKNAMES_PROPORTION}. "
+            f"Noise level has been adjusted to {1/metadata.NICKNAMES_PROPORTION}."
+        )
+
+
+DEFAULT_PARAMETER_CONFIG_VALIDATOR_MAP = {
```

Comment:
nit: move this to the top

Reply:
It doesnt allow that since the functions are defined below it when it is at the top.